auth-api/src/auth_api/services/api_gateway.py

```diff
@@ -74,20 +81,41 @@ def create_key(cls, org_id: int, request_json: Dict[str, str]):
             )
             response = api_key_response.json()
 
+        cls._create_user_and_membership_for_api_user(org_id, env)
```

Comment:
still not very clear about sandbox environment work. but looking at this method - create_key - do we need to call it twice: prod/sandbox, in which case do we update the membership row only the first time create_key is called? and then am i correct to assume that regardless of when the method is called (during prod or sandbox api key generation) the membership row is created in prod db (sandbox db will have a different mechanism to sync this membership)

Reply:
Yes, it needs to be called twice, once for PROD (PROD will fire over the ORG to SANDBOX) then once for SANDBOX

They have two separate clients in keycloak, thus two separate service accounts.